src/promptflow/promptflow/_sdk/entities/_experiment.py

```diff
@@ -188,6 +189,46 @@ def _save_snapshot(self, target):
         self.code = saved_path.resolve().absolute().as_posix()
 
 
+class ChatGroupNode(YAMLTranslatableMixin):
+    def __init__(
+        self,
+        name,
+        roles: List[Dict[str, Any]],
+        max_turns: Optional[int] = None,
+        max_tokens: Optional[int] = None,
+        max_time: Optional[int] = None,
+        stop_signal: Optional[str] = None,
+        **kwargs,
+    ):
+        self.type = ExperimentNodeType.CHAT_GROUP
+        self.name = name
+        self.roles = roles
+        self.max_turns = max_turns
+        self.max_tokens = max_tokens
+        self.max_time = max_time
+        self.stop_signal = stop_signal
+
+    @classmethod
+    def _get_schema_cls(cls):
+        return ChatGroupSchema
+
+    def _save_snapshot(self, target):
+        """Save chat group source to experiment snapshot."""
+        target = Path(target).resolve()
+        logger.debug(f"Saving chat group node {self.name!r} snapshot to {target.as_posix()!r}.")
+        saved_path = target / self.name
+        saved_path.mkdir(parents=True, exist_ok=True)
+        for role in self.roles:
+            role_path = Path(role["path"]).resolve()
+            if not role_path.exists():
+                raise ExperimentValueError(f"Chat role path {role_path.as_posix()!r} does not exist.")
+
+            if role_path.is_dir():
+                shutil.copytree(src=role_path, dst=saved_path / role["role"])
```

Comment:
do we have some common utility here? shall we handle advanced feature like addtional_includes here?

Reply:
Think we can do it once we have multiple places need this functionality. For additional includes ask, good point i have created a task to track.